### DISCUSSION TITLE
Fetching with query without deserializing, JS/TS

### DISCUSSION THREAD

**Discussion by CYihan** [flatgeobuf/flatgeobuf#143] [Q&A] [answered]
Hi,

Is it possible to fetch with a query from a fgb file without deserializing it to another format? 

Our problem is that we want a backend to handle the querying and only send the front end the queried data. We dont want to allow access to the whole fgb. So my idea is that the backend fetches with query and then it sends to front end without deserializing. The front end will then deserializes the queried data into geojson.

I have been looking at the example and the source code for JS/TS but not being able to figure out this question.

**Comment by bjornharrtell** [comment, marked as answer]
Yes. This should be clear from the three linked examples at https://flatgeobuf.org/. Can you explain more specifically what is unclear or what information you think is missing?

**Reply by CYihan** [reply]
Thank you for your quick reply. My apologies, i might have been a bit unclear in my first message.

From the example leaflet/filtered.html:
`flatgeobuf.deserialize('/test/data/UScounties.fgb', fgBoundingBox(), handleHeaderMeta)`

As per my understanding, the above code feches a sub-set of data from the larger set '/test/data/UScounties.fgb'.
Only the sub-set is sent to the client. The client recieves this sub-set as a buffer and converts it to geojson (correct?).

I would like to fetch a sub-set and not convert it to geojson after i've fetched it but remain as a sub-set buffer it is sent as. Is that possible?

Thanks in advance.

**Reply by bjornharrtell** [reply]
Ah ok I think I understand now. Yes it is possible. Under the high level API, the filtered query receives a set of raw binary chunks. You can use the more low level generic API to handle the decoding step yourself. See https://unpkg.com/flatgeobuf@3.18.0/dist/doc/modules/generic.html#deserialize.

**Reply by bjornharrtell** [reply]
Documentation is lacking here so you will need to dig into the source to fully understand it. You can see here https://github.com/flatgeobuf/flatgeobuf/blob/70d5cbf9c07d247bbe842d3f93e2bddf0ff26ae3/src/ts/generic/featurecollection.ts#L146 that we have the raw bytes there and `Feature.getRootAsFeature(bb)` gives you the raw flatbuffers API for the Feature schema to access the data directly.

**Reply by bjornharrtell** [reply]
One important detail that might not be apparent until you dig into the source is that a filtered query against a FlatGeobuf doesn't give a single FlatGeobuf buffer back - it will give you the contained individual Feature buffers because they are not necessarily contiguous in the original file. You could assemble them back into a non-indexed FlatGeobuf by writing a header and concatenating the fetched bytes but I'm not sure why you would want to do that.

**Reply by CYihan** [reply]
Thank you for the answers. I do believe it is enough to get individual feature buffers.

**Reply by jdoig** [reply]
I don't suppose anyone has done similar, and has some code they could link here showing this in action? (i.e: creating a sub-flatgeobuf file from a rect filter?)
